source/remoteClient/localMachine.py

```diff
@@ -0,0 +1,314 @@
+		"""
+		if (
+			self.receivingBraille
+			and braille.handler.displaySize > 0
+			and len(cells) <= braille.handler.displaySize
+		):
+			cells = cells + [0] * (braille.handler.displaySize - len(cells))
+			wx.CallAfter(braille.handler._writeCells, cells)
+
+	def brailleInput(self, **kwargs: Dict[str, Any]) -> None:
+		"""Process braille input gestures from a remote machine.
+
+		Executes braille input commands locally using NVDA's input gesture system.
+		Handles both display routing and braille keyboard input.
+
+		Args:
+			**kwargs: Gesture parameters passed to BrailleInputGesture
+
+		Note:
+			Silently ignores gestures that have no associated action.
+		"""
+		try:
+			inputCore.manager.executeGesture(input.BrailleInputGesture(**kwargs))
+		except inputCore.NoInputGestureAction:
+			pass
+
+	def setBrailleDisplay_size(self, sizes: List[int]) -> None:
+		"""Cache remote braille display sizes for size negotiation.
+
+		Args:
+			sizes: List of display sizes (cells) from remote machines
+		"""
+		self._cachedSizes = sizes
+
+	def handleFilterDisplaySize(self, value: int) -> int:
+		"""Filter the local display size based on remote display sizes.
+
+		Determines the optimal display size when sharing braille output by
+		finding the smallest positive size among local and remote displays.
+
+		Args:
+			value: Local display size in cells
+
+		Returns:
+			int: The negotiated display size to use
+		"""
+		if not self._cachedSizes:
+			return value
+		sizes = self._cachedSizes + [value]
+		try:
+			return min(i for i in sizes if i > 0)
+		except ValueError:
+			return value
+
+	def handleDecideEnabled(self) -> bool:
+		"""Determine if the local braille display should be enabled.
+
+		Returns:
+			bool: False if receiving remote braille, True otherwise
+		"""
+		return not self.receivingBraille
+
+	def sendKey(
+		self,
+		vk_code: Optional[int] = None,
+		extended: Optional[bool] = None,
+		pressed: Optional[bool] = None,
+	) -> None:
+		"""Simulate a keyboard event on the local machine.
+
+		Args:
+			vk_code: Virtual key code to simulate
+			extended: Whether this is an extended key
+			pressed: True for key press, False for key release
+		"""
+		wx.CallAfter(input.sendKey, vk_code, None, extended, pressed)
+
+	def setClipboardText(self, text: str) -> None:
+		"""Set the local clipboard text from a remote machine.
+
+		Args:
+			text: Text to copy to the clipboard
+			**kwargs: Additional parameters (ignored for compatibility)
+		"""
+		cues.clipboard_received()
+		api.copyToClip(text=text)
+
+	def sendSAS(self) -> None:
+		"""
+		Simulate a secure attention sequence (e.g. CTRL+ALT+DEL).
+
+		SendSAS requires UI Access. If this fails, a warning is displayed.
+		"""
+		if hasUiAccess():
+			ctypes.windll.sas.SendSAS(0)
+		else:
+			# Translators: Message displayed when a remote machine tries to send a SAS but UI Access is disabled.
+			ui.message(_("No permission on device to trigger CTRL+ALT+DEL from remote"))
```

Comment:
Can these be made a bit more user friendly?
```suggestion
			# Translators: Message displayed when a remote machine tries to send alt+ctrl+delete but the local copy of NVDA does not have adequate permissions.
			ui.message(_("Unable to trigger control alt delete from remote"))
```